### PR TITLE
Parsing only the authority when the request is a CONNECT. #5252

### PR DESCRIPTION
**Pull request by GpMits**
Addressing #5252

### CODE REVIEW

File: src/org/parosproxy/paros/network/HttpRequestHeader.java [hunk 1]
```diff
@@ -449,29 +449,30 @@ private void parse(boolean isSecure) throws URIException, HttpMalformedHeaderExc
...
-        }
-
-        if (mUri.getScheme().equalsIgnoreCase(HTTPS)) {
-            setSecure(true);
-        }
-
-        String hostHeader;
         if (mMethod.equalsIgnoreCase(CONNECT)) {
+            String hostHeader;
             hostHeader = sUri;
```
Comment (kingthorin) [inline src/org/parosproxy/paros/network/HttpRequestHeader.java]: Just do this on one line? `String hostHeader = sUri;`

Comment (GpMits) [reply src/org/parosproxy/paros/network/HttpRequestHeader.java]: Sorry, I didn't notice this one. Fixing...

Comment (thc202) [reply src/org/parosproxy/paros/network/HttpRequestHeader.java]: The `hostHeader` can be removed and `sUri` used directly though.

**Comment by thc202** [comment]
The commit email is not associated with your account (in case you want to change that).

Changes to Paros classes should have a ZAP comment:
https://github.com/zaproxy/zaproxy/wiki/DevGuidelines

File: src/org/parosproxy/paros/network/HttpRequestHeader.java (hunk 1 above)
Comment (kingthorin) [reply src/org/parosproxy/paros/network/HttpRequestHeader.java]: DOH, sorry I didn't pick that up.

Comment (GpMits) [reply src/org/parosproxy/paros/network/HttpRequestHeader.java]: Thanks for the heads up! Fixed!

**Comment by GpMits** [comment]
@thc202 Thanks for the heads up! Fixed!

File: src/org/parosproxy/paros/network/HttpRequestHeader.java [hunk 2]
```diff
@@ -449,29 +450,29 @@ private void parse(boolean isSecure) throws URIException, HttpMalformedHeaderExc
             throw new HttpMalformedHeaderException("Unexpected version: " + mVersion);
         }
 
-        mUri = parseURI(sUri);
+        // ZAP: Only parsing the authority on CONNECT requests. Required to handle URIs with underscore
```
Comment (thc202) [inline src/org/parosproxy/paros/network/HttpRequestHeader.java]: This can be removed, it's fine adding just the ZAP comment at the top.

**Comment by thc202** [comment]
Thank you!

**Comment by kingthorin** [comment]
Squashed and merged. Thanks!

**Comment by thc202** [comment]
@GpMits how would you like to be credited?
https://github.com/zaproxy/zap-core-help/wiki/HelpCredits

**Comment by kingthorin** [comment]
@GpMits how would you like to be credited? https://github.com/zaproxy/zap-core-help/wiki/HelpCredits#zap-extended-team

**Comment by kingthorin** [comment]
Hahahaha oops

**Comment by thc202** [comment]
:)

**Comment by GpMits** [comment]
Thank you ! @thc202 @kingthorin just my name (Gustavo Mitsuichi) is fine :)

**Pull request merged**